modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/ghoul.dm

```diff
@@ -0,0 +1,282 @@
+	inherent_traits = list(TRAIT_ADVANCEDTOOLUSER,
+		TRAIT_RADIMMUNE,
+		TRAIT_CAN_STRIP,
+		TRAIT_EASYDISMEMBER,
+	)
+	offset_features = list(OFFSET_UNIFORM = list(0,0), OFFSET_ID = list(0,0), OFFSET_GLOVES = list(0,0), OFFSET_GLASSES = list(0,1), OFFSET_EARS = list(0,1), OFFSET_SHOES = list(0,0), \
+						   OFFSET_S_STORE = list(0,0), OFFSET_FACEMASK = list(0,1), OFFSET_HEAD = list(0,1), OFFSET_FACE = list(0,1), OFFSET_BELT = list(0,0), OFFSET_BACK = list(0,0), \
+						   OFFSET_SUIT = list(0,0), OFFSET_NECK = list(0,1))
+	toxic_food = DAIRY | PINEAPPLE
+	disliked_food = VEGETABLES | FRUIT | CLOTH
+	liked_food = RAW | MEAT
+	payday_modifier = 0.75 //-- "Equality"
+	speedmod = -0.35 //speedy mfer
+	armor = -100 //2x more damage
+	stunmod = 1.25 //multiplier for stun durations
+	punchdamagelow = 1 //lowest possible punch damage. if this is set to 0, punches will always miss
+	punchdamagehigh = 5 //highest possible punch damage
+	siemens_coeff = 0.7 //base electrocution coefficient
+	bodytemp_normal = T20C
+	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_MAGIC | MIRROR_PRIDE | ERT_SPAWN | RACE_SWAP | SLIME_EXTRACT
+	limbs_icon = 'modular_skyrat/master_files/icons/mob/species/ghoul_bodyparts.dmi'
+
+	var/dehydrate = 0
+
+	bodypart_overides = list(
+	BODY_ZONE_L_ARM = /obj/item/bodypart/l_arm/ghoul,\
+	BODY_ZONE_R_ARM = /obj/item/bodypart/r_arm/ghoul,\
+	BODY_ZONE_HEAD = /obj/item/bodypart/head/ghoul,\
+	BODY_ZONE_L_LEG = /obj/item/bodypart/l_leg/ghoul,\
+	BODY_ZONE_R_LEG = /obj/item/bodypart/r_leg/ghoul,\
+	BODY_ZONE_CHEST = /obj/item/bodypart/chest/ghoul)
+
+/proc/proof_ghoul_features(list/inFeatures)
+	// Missing Defaults in DNA? Randomize!
+	if(inFeatures["ghoulcolor"] == null || inFeatures["ghoulcolor"] == "")
+		inFeatures["ghoulcolor"] = GLOB.color_list_ghoul[pick(GLOB.color_list_ghoul)]
+
+/datum/species/ghoul/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+	// Missing Defaults in DNA? Randomize!
+	proof_ghoul_features(C.dna.features)
+
+	. = ..()
+
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+
+		set_ghoul_color(H)
+
+		// 2) BODYPARTS
+		C.part_default_head = /obj/item/bodypart/head/ghoul
+		C.part_default_chest = /obj/item/bodypart/chest/ghoul
+		C.part_default_l_arm = /obj/item/bodypart/l_arm/ghoul
+		C.part_default_r_arm = /obj/item/bodypart/r_arm/ghoul
+		C.part_default_l_leg = /obj/item/bodypart/l_leg/ghoul
+		C.part_default_r_leg = /obj/item/bodypart/r_leg/ghoul
+		C.ReassignForeignBodyparts()
+
+/datum/species/proc/set_ghoul_color(mob/living/carbon/human/H)
+	return // Do Nothing
+
+/datum/species/ghoul/set_ghoul_color(mob/living/carbon/human/H)
+	// Called on Assign, or on Color Change (or any time proof_ghoul_features() is used)
+	fixed_mut_color = H.dna.features["ghoulcolor"]
+	default_color = fixed_mut_color
+
+/mob/living/carbon/proc/ReassignForeignBodyparts()
+	var/obj/item/bodypart/head = get_bodypart(BODY_ZONE_HEAD)
+	if (head?.type != part_default_head)
+		qdel(head)
```

Comment:
This might end up qdel'ing their brain too, not sure though.

Reply:
it seems to work fine for me